### ISSUE TITLE
VSC freezes when using URL's in config file

### ISSUE THREAD

**Issue by EpiqSty** [codeout/vscode-junos#2]
Looks like this project has inherited bug from the upstream:
https://github.com/woodjme/vscode-junos-syntax/issues/1
i.e in case of any valid URL, VSCode crashes.
Do you  have any ideas how to fix/prevent it?

**Comment by EpiqSty** [comment]
As the workaround, I've removed whole http section from `~\.vscode\extensions\codeout.vscode-junos-0.1.3\vscode-junos-syntax\syntaxes\Junos.tmLanguage` : lines from 215 to 222:
```
		<dict>
			<key>comment</key>
			<string>URL Addresses, either matching character URL patterns, or http(s):// followed by an IPv4 address. </string>
			<key>match</key>
			<string>\b((https?:)|(scp:)|((s?|t?)ftp:))//((([\da-zA-Z\.-]+)\.([A-Za-z\.]{2,6}))|(\b((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))\b))(([/\w\.&amp;?=+-]*)*)/?</string>
			<key>name</key>
			<string>keyword.operator.junos</string>
		</dict>
```
And after VSCode restart I was able to open Junos config file containing URL without application crash.

I am not completely sure, what exactly is wrong with that section, but maybe regexp is incorrect? https://regex101.com/ showing me "pattern error" for PCRE(PHP) and JavaScript, but it's ok for Python&Golang. Which type of regular expressions is eligible to use in VSCode extensions?

**Comment by codeout** [comment]
Thanks for your crash report,

Yes, the regexp might be incorrect. Each `/` should be `\/`.
Could you help to try correct one and see if VSCode still crashes?

```diff
- \b((https?:)|(scp:)|((s?|t?)ftp:))//((([\da-zA-Z\.-]+)\.([A-Za-z\.]{2,6}))|(\b((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))\b))(([/\w\.&amp;?=+-]*)*)/?
+ \b((https?:)|(scp:)|((s?|t?)ftp:))\/\/((([\da-zA-Z\.-]+)\.([A-Za-z\.]{2,6}))|(\b((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]    {2}|2[0-4][0-9]|25[0-5]))\b))(([\/\w\.&amp;?=+-]*)*)\/?
```

You can test here: https://regex101.com/r/FyLGcF/1

It doesn't freeze even with incorrect one in my environment actually, so I have no idea about the root cause so far.

**Comment by EpiqSty** [comment]
@codeout , I've tried with adjusted regexp, but it did not helped.
The behaviour exactly the same, in case of any URL appearing in the file, VSCode is getting stuck completely:

![image](https://user-images.githubusercontent.com/25151779/53323922-be1dd780-38df-11e9-81b7-389794e67d7c.png)

**Comment by codeout** [comment]
Thanks. Then we need to patch as you pointed out until the upstream is fixed.

**Comment by codeout** [comment]
Just published 0.1.4 with your workaround. Luckily, we don't see any significant change here.
( `ae1` below looks storage a bit but acceptable, I think )

## 0.1.3

![image](https://user-images.githubusercontent.com/921696/53692628-b4600c80-3dd6-11e9-88c8-65275a104355.png)

## 0.1.4

![image](https://user-images.githubusercontent.com/921696/53692629-bcb84780-3dd6-11e9-8959-75aee08881e7.png)

**Comment by EpiqSty** [comment]
@codeout , i've updated  extension , all fine. Thanks a lot!
